lib/repositories/mailbox.js

```diff
@@ -82,6 +82,49 @@ function createApi(table, provider, sqlGenerator, dependencies) {
     },
 
     /**
+     * Returns an array of all mailboxes belonging to the specified context
+     */
+    findByContext: function(context) {
+      dependencies.logger.trace('mailbox.findByContext called');
+
+      var query = table
+        .select(table.star())
+        .from(table)
+        .where(table['context_id'].equals(context.getId()))
+        .toQuery();
+
+      return common.find(query, provider, constructor.bind(this))
+        .then(function(result) {
+          return result.reduce(function(mailboxes, mailbox) {
+            mailboxes.push(mailbox);
+            return mailboxes;
+          }, []);
+        });
+
+      function constructor(id) {
+        /*jshint validthis:true*/
+        //Passing undefined so that number can be populated by db result
+        return this.create(undefined, context, {}, id);
```

Comment:
The third param can either be an object representing fields or an id. You can just remove that parameter, it's intended to work that way.